packages/fxa-auth-server/lib/l10n/index.ts

```diff
@@ -91,26 +95,35 @@ class Localizer {
    * @param locale Locale to use, defaults to en.
    */
   protected async fetchTranslatedMessages(locale?: string) {
-    // note: 'en' auth.ftl only exists for browser bindings / Storybook
-    // the fallback English strings within the templates will be shown in other envs
+    // Note: 'en' auth.ftl only exists for browser bindings / Storybook. The fallback
+    // English strings within the templates are tested and are shown in other envs
     const path = `${this.bindings.opts.translations.basePath}/${
       locale || 'en'
     }/auth.ftl`;
 
     return this.bindings.fetchResource(path);
   }
 
-  // NOTE: not currently used, will be implemented in FXA-4623
-  protected async localizeString(
-    acceptLanguage: string,
-    { id, message }: FtlIdMsg
-  ) {
+  async localizeStrings(
+    acceptLanguage = 'en',
+    ftlIdMsgs: FtlIdMsg[]
+  ): Promise<LocalizedStrings> {
     const { l10n } = await this.setupLocalizer(acceptLanguage);
-    return (await l10n.formatValue(id, message)) || message;
+
+    const localizedFtlIdMsgs = await Promise.all(
+      ftlIdMsgs.map(async (ftlIdMsg) => {
+        const { id, message } = ftlIdMsg;
+        return Promise.resolve({
+          [id]: (await l10n.formatValue(id, message)) || message,
```

Comment:
Perhaps an error here should be handled? As we've been discussing somewhat recently, an unhandled error will cause promsie.all to reject immediately. It seems like one msg failing to format shouldn't prevent the others from being processed.

Reply:
In this case, I don't believe it can reject. If `await l10n.formatValue(id, message)` rejects, the promise should get resolved with the original `message`. Good thinking though, I'll write a test for this to verify it's doing what I think it is.